apps/common/src/main/scala/org/lfdecentralizedtrust/splice/automation/SqlIndexInitializationTrigger.scala

```diff
@@ -0,0 +1,316 @@
+      _ = if (unexpected.nonEmpty) {
+        logger.debug(s"Found SQL indexes: $actual")
+        logger.warn(s"Found unexpected SQL indexes: ${unexpected.mkString(", ")}. ")
+      }
+      _ <- MonadUtil.sequentialTraverse(expectedIndexes.toSeq) {
+        case (indexName, IndexAction.ManagedByFlyway) =>
+          if (!actual.contains(indexName)) {
+            logger.warn(
+              s"Index $indexName is expected to have been created by Flyway, but was not found"
+            )
+          }
+          FutureUnlessShutdown.unit
+        case (indexName, IndexAction.Drop) =>
+          logger.info(s"Dropping index $indexName")
+          storage.update(sqlu"drop index concurrently if exists #$indexName", indexName)
+        case (indexName, IndexAction.Create(action)) =>
+          if (!actual.contains(indexName)) {
+            logger.info(s"Creating index $indexName")
+            storage
+              .update(action, indexName)
+              .thereafter(_ => logger.info(s"Finished creating index $indexName"))
+          } else {
+            FutureUnlessShutdown.unit
+          }
+      }
+    } yield ()
+  }
+}
+
+object SqlIndexInitializationTrigger {
+
+  def apply(
+      storage: Storage,
+      triggerContext: TriggerContext,
+      expectedIndexes: Map[String, IndexAction] = defaultExpectedIndexes,
+  )(implicit ec: ExecutionContext, tracer: Tracer): SqlIndexInitializationTrigger = storage match {
+    case dbStorage: DbStorage =>
+      new SqlIndexInitializationTrigger(
+        dbStorage,
+        triggerContext,
+        expectedIndexes,
+      )
+    case storageType =>
+      // Same behavior as in `ScanStore.apply` and similar - we only really support DbStorage in our apps.
+      throw new RuntimeException(s"Unsupported storage type $storageType")
+  }
+
+  final case class IndexData(
+      tableName: String,
+      definition: String,
+  ) extends PrettyPrinting {
+    override def pretty: Pretty[this.type] = prettyOfClass(
+      param("tableName", _.tableName.doubleQuoted),
+      param("definition", _.definition.doubleQuoted),
+    )
+  }
+  implicit val GetResultIndexData: GetResult[IndexData] =
+    GetResult { prs =>
+      import prs.*
+      (IndexData.apply _).tupled(
+        (
+          <<[String],
+          <<[String],
+        )
+      )
+    }
+
+  sealed trait IndexAction
+  object IndexAction {
+
+    /** Do not touch this index, it is managed by Flyway. */
+    final case object ManagedByFlyway extends IndexAction
+
+    /** Drop this index if it exists. */
+    final case object Drop extends IndexAction
+
+    /** Create this index if it does not exist. */
+    final case class Create(action: DBIOAction[?, NoStream, Effect.Write & Effect.Transactional])
+        extends IndexAction
+  }
+
+  /** All indexes created by Flyway after applying all migrations in `apps/common/src/main/resources/db/migration/canton-network/postgres/stable` */
+  val flywayIndexes: Map[String, IndexAction] = Map(
```

Comment:
Would it not be possible to parse the .sql files from flyway to get these names? This is a lot of extra admin I would like to avoid, easy to forget.